clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -777,6 +783,38 @@ constructKernelName(Sema &S, FunctionDecl *KernelCallerFunc,
                                       KernelNameType)};
 }
 
+static bool isDefaultSPIRArch(ASTContext &Context) {
+  llvm::Triple T = Context.getTargetInfo().getTriple();
+  if (T.isSPIR() && T.getSubArch() == llvm::Triple::NoSubArch)
+    return true;
+  return false;
+}
+
+static ParmVarDecl *getSyclKernelHandlerArg(FunctionDecl *KernelCallerFunc) {
+  // Specialization constants in SYCL 2020 are not captured by lambda and
+  // accessed through new optional lambda argument kernel_handler
+  auto KHArg =
+      std::find_if(KernelCallerFunc->param_begin(),
```

Comment:
```suggestion
      llvm::find_if(KernelCallerFunc->parameters(),
```

No param_end below either.